cmd/skopeo/inspect.go

```diff
@@ -39,42 +38,42 @@ type inspectOptions struct {
 	config bool // Output the raw config blob instead of parsing information about the image
 }
 
-func inspectCmd(global *globalOptions) cli.Command {
+func inspectCommand(globalFlagResults *globalOptions) *cobra.Command {
+	var (
+		opts               inspectOptions
+		inspectDescription = fmt.Sprintf(`Return low-level information about "IMAGE-NAME" in a registry/transport
+Supported transports:
+%s
+
+See skopeo(1) section "IMAGE NAMES" for the expected format
+`, strings.Join(transports.ListNames(), ", "))
+	)
+	inspectCommand := &cobra.Command{
+		Use:   "inspect [command options] IMAGE-NAME",
+		Short: "Inspect image IMAGE-NAME",
+		Long:  inspectDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return inspectCmd(cmd, args, opts)
+		},
+		Example: `skopeo inspect docker://docker.io/fedora`,
+	}
+	inspectCommand.SetUsageTemplate(usageTemplate())
+	inspectCommand.DisableFlagsInUseLine = true
+
+	flags := inspectCommand.Flags()
+	flags.BoolVar(&opts.raw, "raw", false, "output raw manifest or configuration")
+	flags.BoolVar(&opts.config, "config", false, "output configuration")
 	sharedFlags, sharedOpts := sharedImageFlags()
-	imageFlags, imageOpts := imageFlags(global, sharedOpts, "", "")
-	opts := inspectOptions{
-		global: global,
-		image:  imageOpts,
-	}
-	return cli.Command{
-		Name:  "inspect",
-		Usage: "Inspect image IMAGE-NAME",
-		Description: fmt.Sprintf(`
-	Return low-level information about "IMAGE-NAME" in a registry/transport
-
-	Supported transports:
-	%s
-
-	See skopeo(1) section "IMAGE NAMES" for the expected format
-	`, strings.Join(transports.ListNames(), ", ")),
-		ArgsUsage: "IMAGE-NAME",
-		Flags: append(append([]cli.Flag{
-			cli.BoolFlag{
-				Name:        "raw",
-				Usage:       "output raw manifest or configuration",
-				Destination: &opts.raw,
-			},
-			cli.BoolFlag{
-				Name:        "config",
-				Usage:       "output configuration",
-				Destination: &opts.config,
-			},
-		}, sharedFlags...), imageFlags...),
-		Action: commandAction(opts.run),
-	}
+	imageFlags, imageOpts := imageFlags(globalFlagResults, sharedOpts, "", "")
+	opts.global = globalFlagResults
+	opts.image = imageOpts
+	flags.AddFlagSet(&sharedFlags)
+	flags.AddFlagSet(&imageFlags)
+	return inspectCommand
 }
 
-func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error) {
+func inspectCmd(c *cobra.Command, args []string, opts inspectOptions) (retErr error) {
+	stdout := c.OutOrStdout()
```

Comment:
This should be done in `commandAction` or something like that, not something every single command should have to worry about. (See the other comments about `commandAction` as well.)